lib/md039.mjs

```diff
@@ -35,18 +36,6 @@ function addLabelSpaceError(onError, label, labelText, isStart) {
   );
 }
 
-/**
- * Determines if a link is a valid link (and not a fake shortcut link due to parser tricks).
- *
- * @param {import("markdownlint").MicromarkToken} label Label token.
- * @param {import("markdownlint").MicromarkToken} labelText LabelText token.
- * @param {Map<string, any>} definitions Map of link definitions.
- * @returns {boolean} True iff the link is valid.
- */
-function validLink(label, labelText, definitions) {
-  return (label.parent?.children.length !== 1) || definitions.has(labelText.text.trim());
-}
-
 /** @type {import("markdownlint").Rule} */
 export default {
   "names": [ "MD039", "no-space-in-links" ],
```

Comment:
I closely copied this rule's logic since it looked similar to what we're trying to (aka parse the link text).
I ended up moving this `validLink` to a shared helper.